### ISSUE TITLE
[v2 BUG]: no features generators in predict.py

### ISSUE THREAD

**Issue by jsanders802** [chemprop/chemprop#929]
**Describe the bug**
I receive errors when using a model trained with --features-generators morgan_binary to make predictions.

**Example(s)**
chemprop predict --model-path chemprop/log_property_morganbin/ --test-path smiles.csv --preds-path predictions.csv --features-generators morgan_binary

**Expected behavior**
I expected to get an output file with predictions (as I do with a model trained without using Morgan fingerprints).  I see there was another bug for including this for hpopt...

**Error Stack Trace**

```
/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/torch/cuda/__init__.py:619: UserWarning: Can't initialize NVML
  warnings.warn("Can't initialize NVML")
/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/torch/cuda/__init__.py:749: UserWarning: CUDA initialization: The NVIDIA driver on your system is too old (found version 11040). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org to install a PyTorch version that has been compiled with your version of the CUDA driver. (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)
  return torch._C._cuda_getDeviceCount() if nvml_count < 0 else nvml_count
/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/utilities/parsing.py:199: Attribute 'graph_transform' is an instance of `nn.Module` and is already saved during checkpointing. It is recommended to ignore them using `self.save_hyperparameters(ignore=['graph_transform'])`.
/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/lightning/pytorch/utilities/parsing.py:199: Attribute 'X_d_transform' is an instance of `nn.Module` and is already saved during checkpointing. It is recommended to ignore them using `self.save_hyperparameters(ignore=['X_d_transform'])`.
Traceback (most recent call last):
  File "/home/jsanders/.conda/envs/chemprop/bin/chemprop", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/main.py", line 80, in main
    func(args)
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/predict.py", line 34, in func
    main(args)
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/predict.py", line 315, in main
    make_prediction_for_model(args, model_path, multicomponent, output_path)
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/predict.py", line 212, in make_prediction_for_model
    test_data = build_data_from_files(
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/utils/parsing.py", line 288, in build_data_from_files
    mol_data, rxn_data = make_datapoints(
                         ^^^^^^^^^^^^^^^^
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/utils/parsing.py", line 210, in make_datapoints
    mol_data = [
               ^
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/utils/parsing.py", line 211, in <listcomp>
    [
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/cli/utils/parsing.py", line 212, in <listcomp>
    MoleculeDatapoint.from_smi(
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/data/datapoints.py", line 65, in from_smi
    return cls(mol, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 14, in __init__
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/data/datapoints.py", line 98, in __post_init__
    super().__post_init__(mfs)
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/data/datapoints.py", line 41, in __post_init__
    self.x_d = self.calc_features(mfs)
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/data/datapoints.py", line 107, in calc_features
    return np.hstack([mf(self.mol) for mf in mfs])
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jsanders/.conda/envs/chemprop/lib/python3.11/site-packages/chemprop/data/datapoints.py", line 107, in <listcomp>
    return np.hstack([mf(self.mol) for mf in mfs])
                      ^^^^^^^^^^^^
TypeError: 'str' object is not callable
```

**Screenshots**

**Environment**
- python version 3.11
- package versions: `conda list` or `pip list`

```
# packages in environment at /home/jsanders/.conda/envs/chemprop:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             5.1                       1_gnu
aimsim-core               2.2.1                    pypi_0    pypi
aiohttp                   3.9.5                    pypi_0    pypi
aiohttp-cors              0.7.0                    pypi_0    pypi
aiosignal                 1.3.1                    pypi_0    pypi
annotated-types           0.7.0                    pypi_0    pypi
anyio                     4.4.0                    pypi_0    pypi
astartes                  1.2.2                    pypi_0    pypi
attrs                     23.2.0                   pypi_0    pypi
bzip2                     1.0.8                h5eee18b_6
ca-certificates           2024.3.11            h06a4308_0
cachetools                5.3.3                    pypi_0    pypi
certifi                   2024.6.2                 pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
chemprop                  2.0.2                    pypi_0    pypi
click                     8.1.7                    pypi_0    pypi
cloudpickle               3.0.0                    pypi_0    pypi
colorful                  0.5.6                    pypi_0    pypi
configargparse            1.7                      pypi_0    pypi
dill                      0.3.8                    pypi_0    pypi
distlib                   0.3.8                    pypi_0    pypi
dnspython                 2.6.1                    pypi_0    pypi
email-validator           2.1.1                    pypi_0    pypi
fastapi                   0.111.0                  pypi_0    pypi
fastapi-cli               0.0.4                    pypi_0    pypi
filelock                  3.14.0                   pypi_0    pypi
frozenlist                1.4.1                    pypi_0    pypi
fsspec                    2024.6.0                 pypi_0    pypi
future                    1.0.0                    pypi_0    pypi
google-api-core           2.19.0                   pypi_0    pypi
google-auth               2.30.0                   pypi_0    pypi
googleapis-common-protos  1.63.1                   pypi_0    pypi
grpcio                    1.64.1                   pypi_0    pypi
h11                       0.14.0                   pypi_0    pypi
httpcore                  1.0.5                    pypi_0    pypi
httptools                 0.6.1                    pypi_0    pypi
httpx                     0.27.0                   pypi_0    pypi
hyperopt                  0.2.7                    pypi_0    pypi
idna                      3.7                      pypi_0    pypi
jinja2                    3.1.4                    pypi_0    pypi
joblib                    1.4.2                    pypi_0    pypi
jsonschema                4.22.0                   pypi_0    pypi
jsonschema-specifications 2023.12.1                pypi_0    pypi
ld_impl_linux-64          2.38                 h1181459_1
libffi                    3.4.4                h6a678d5_1
libgcc-ng                 11.2.0               h1234567_1
libgomp                   11.2.0               h1234567_1
libstdcxx-ng              11.2.0               h1234567_1
libuuid                   1.41.5               h5eee18b_0
lightning                 2.2.5                    pypi_0    pypi
lightning-utilities       0.11.2                   pypi_0    pypi
linkify-it-py             2.0.3                    pypi_0    pypi
markdown-it-py            3.0.0                    pypi_0    pypi
markupsafe                2.1.5                    pypi_0    pypi
mdit-py-plugins           0.4.1                    pypi_0    pypi
mdurl                     0.1.2                    pypi_0    pypi
memray                    1.12.0                   pypi_0    pypi
mhfp                      1.9.6                    pypi_0    pypi
mordredcommunity          2.0.5                    pypi_0    pypi
mpmath                    1.3.0                    pypi_0    pypi
msgpack                   1.0.8                    pypi_0    pypi
multidict                 6.0.5                    pypi_0    pypi
multiprocess              0.70.16                  pypi_0    pypi
ncurses                   6.4                  h6a678d5_0
networkx                  3.3                      pypi_0    pypi
numpy                     1.26.4                   pypi_0    pypi
nvidia-cublas-cu12        12.1.3.1                 pypi_0    pypi
nvidia-cuda-cupti-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-nvrtc-cu12    12.1.105                 pypi_0    pypi
nvidia-cuda-runtime-cu12  12.1.105                 pypi_0    pypi
nvidia-cudnn-cu12         8.9.2.26                 pypi_0    pypi
nvidia-cufft-cu12         11.0.2.54                pypi_0    pypi
nvidia-curand-cu12        10.3.2.106               pypi_0    pypi
nvidia-cusolver-cu12      11.4.5.107               pypi_0    pypi
nvidia-cusparse-cu12      12.1.0.106               pypi_0    pypi
nvidia-nccl-cu12          2.20.5                   pypi_0    pypi
nvidia-nvjitlink-cu12     12.5.40                  pypi_0    pypi
nvidia-nvtx-cu12          12.1.105                 pypi_0    pypi
opencensus                0.11.4                   pypi_0    pypi
opencensus-context        0.1.3                    pypi_0    pypi
openssl                   3.0.13               h7f8727e_2
orjson                    3.10.3                   pypi_0    pypi
packaging                 24.0                     pypi_0    pypi
padelpy                   0.1.16                   pypi_0    pypi
pandas                    2.2.2                    pypi_0    pypi
pillow                    10.3.0                   pypi_0    pypi
pip                       24.0            py311h06a4308_0
platformdirs              4.2.2                    pypi_0    pypi
prometheus-client         0.20.0                   pypi_0    pypi
proto-plus                1.23.0                   pypi_0    pypi
protobuf                  4.25.3                   pypi_0    pypi
psutil                    5.9.8                    pypi_0    pypi
py-spy                    0.3.14                   pypi_0    pypi
py4j                      0.10.9.7                 pypi_0    pypi
pyarrow                   16.1.0                   pypi_0    pypi
pyasn1                    0.6.0                    pypi_0    pypi
pyasn1-modules            0.4.0                    pypi_0    pypi
pydantic                  2.7.3                    pypi_0    pypi
pydantic-core             2.18.4                   pypi_0    pypi
pygments                  2.18.0                   pypi_0    pypi
python                    3.11.9               h955ad1f_0
python-dateutil           2.9.0.post0              pypi_0    pypi
python-dotenv             1.0.1                    pypi_0    pypi
python-multipart          0.0.9                    pypi_0    pypi
pytorch-lightning         2.2.5                    pypi_0    pypi
pytz                      2024.1                   pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
ray                       2.24.0                   pypi_0    pypi
rdkit                     2023.9.6                 pypi_0    pypi
readline                  8.2                  h5eee18b_0
referencing               0.35.1                   pypi_0    pypi
requests                  2.32.3                   pypi_0    pypi
rich                      13.7.1                   pypi_0    pypi
rpds-py                   0.18.1                   pypi_0    pypi
rsa                       4.9                      pypi_0    pypi
scikit-learn              1.5.0                    pypi_0    pypi
scipy                     1.13.1                   pypi_0    pypi
setuptools                69.5.1          py311h06a4308_0
shellingham               1.5.4                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
smart-open                7.0.4                    pypi_0    pypi
sniffio                   1.3.1                    pypi_0    pypi
sqlite                    3.45.3               h5eee18b_0
starlette                 0.37.2                   pypi_0    pypi
sympy                     1.12.1                   pypi_0    pypi
tabulate                  0.9.0                    pypi_0    pypi
tensorboardx              2.6.2.2                  pypi_0    pypi
textual                   0.65.2                   pypi_0    pypi
threadpoolctl             3.5.0                    pypi_0    pypi
tk                        8.6.14               h39e8969_0
torch                     2.3.0                    pypi_0    pypi
torchmetrics              1.4.0.post0              pypi_0    pypi
tqdm                      4.66.4                   pypi_0    pypi
triton                    2.3.0                    pypi_0    pypi
typer                     0.12.3                   pypi_0    pypi
typing-extensions         4.12.1                   pypi_0    pypi
tzdata                    2024.1                   pypi_0    pypi
uc-micro-py               1.0.3                    pypi_0    pypi
ujson                     5.10.0                   pypi_0    pypi
urllib3                   2.2.1                    pypi_0    pypi
uvicorn                   0.30.1                   pypi_0    pypi
uvloop                    0.19.0                   pypi_0    pypi
virtualenv                20.26.2                  pypi_0    pypi
watchfiles                0.22.0                   pypi_0    pypi
websockets                12.0                     pypi_0    pypi
wheel                     0.43.0          py311h06a4308_0
wrapt                     1.16.0                   pypi_0    pypi
xz                        5.4.6                h5eee18b_1
yarl                      1.9.4                    pypi_0    pypi
zlib                      1.2.13               h5eee18b_1
```

- OS   Ubuntu 20.04.6 LTS

**Checklist**
- [ ] all dependencies are satisifed: `conda list` or `pip list` shows the packages listed in the `pyproject.toml`
- [ ] the unit tests are working: `pytest -v` reports no errors

**Additional context**

**Comment by JacksonBurns** [comment]
@hwpang I think you have solved something like this before (maybe), could you take a look?

**Comment by hwpang** [comment]
Thanks for tagging! I make a PR that should fix this: https://github.com/chemprop/chemprop/pull/930.

**Comment by KnathanM** [comment]
An earlier PR #876 would also fix this bug, so we will close #930.
